common/simulation.cpp

```diff
@@ -14,6 +14,30 @@ namespace tec {
 
 double UPDATE_RATE = 1.0 / 8.0; // 8 per second
 double TICKS_PER_SECOND = 60.0 * UPDATE_RATE;
+
+Simulation::Simulation() {
+	worker_thread = new std::thread([this]() {
+		std::unique_lock lock(worker_m);
+		while (this->worker_thread) {
+			worker_cv.wait_for(lock, std::chrono::milliseconds(1));
+			if(!lock.owns_lock()) {
```

Comment:
Where does it get unlocked?

Reply:
the call to wait_for will unlock while waiting to be signaled, then take the lock again before returning, so the thread should be holding the lock when signaled.

Reply:
Oh so the thread sleeps for 1 ms? Should it just wait for the cv to be notified? If you need to watch dog shouldn't it be longer like seconds?